bluetooth: mesh: align sensor/column reporting (#7676)
nrfconnect/sdk-nrf: doc/nrf/libraries/bluetooth_services/mesh/sensor.rst

Sensor series
*************

> Sensor series data may be provided for all sensor types.
> :c:member:`bt_mesh_sensor_series.column_count` must be specified and the sensor series :c:member:`bt_mesh_sensor_series.get` callback must be implemented to enable the sensor's series data feature.
> 
> For sensor types with more than two channels, the series data is organized into a static set of colums, specified at init.

Reviewer: ```suggestion
Sensor series data can be provided for all sensor types.
To enable the sensor's series data feature, :c:member:`bt_mesh_sensor_series.column_count` must be specified and the sensor series :c:member:`bt_mesh_sensor_series.get` callback must be implemented.

For sensor types with more than two channels, the series data is organized into a static set of columns specified at init.
```

The format of the column may be queried with :c:func:`bt_mesh_sensor_column_format_get`.

The ``get`` callback gets called with an index of one of the columns, and is expected to fill the ``value`` parameter with sensor data for the specified column.
If a Sensor Client requests a series of columns, the callback may be called repeatedly, requesting data from each column.

> Example: Three channel sensor (Average ambient temperature in a period of day) as a sensor series:

Reviewer: ```suggestion
Example: A three-channel sensor (average ambient temperature in a period of day) as a sensor series:
```

   static struct sensor_value avg_temp[ARRAY_SIZE(columns)];

   static int getter(struct bt_mesh_sensor *sensor, struct bt_mesh_msg_ctx *ctx,
                     uint32_t column_index, struct sensor_value *value)
   {
       value[0] = avg_temp[column_index];
       value[1] = columns[column_index].start;
       value[2] = columns[column_index].end;

       return 0;
   }

> Example: Single-channel sensor (Motion sensed) as a sensor series:

Reviewer: ```suggestion
Example: Single-channel sensor (motion sensed) as a sensor series:
```
